taf/updater/updater_pipeline.py

```diff
@@ -633,11 +637,11 @@ def clone_repo_to_temp(temp_repo, users_repo):
                     temp_repo.clone_from_disk(
                         users_repo.path,
                         users_repo.get_remote_url(),
-                        is_bare=True,
+                        is_bare=self.bare,
```

Comment:
All repositories which are cloned to temp should always be bare. We do not want to have worktrees before we run validation. So first we clone/move everything to temp, validate, and then clone/update user's repositories. The user's repositories can be bare or not bare.